fppanalysis/time_delay_estimation.py

```diff
@@ -198,14 +199,60 @@ def get_error(params):
 
 
 def estimate_time_delay_ccmax(x: np.ndarray, y: np.ndarray, dt: float):
-    """
-    Estimates the average time delay between to signals by finding the time lag that maximizies the
-    cross-correlation function.
+    """Estimates the average time delay between to signals by finding the time
+    lag that maximizes the cross-correlation function.
+
+    Returns:
+        td Estimated time delay
+        C Cross correlation at a time lag td.
     """
     ccf_times, ccf = cf.corr_fun(x, y, dt=dt, biased=True, norm=True)
     ccf = ccf[np.abs(ccf_times) < max(ccf_times) / 2]
     ccf_times = ccf_times[np.abs(ccf_times) < max(ccf_times) / 2]
-    return ccf_times[np.argmax(ccf)]
+    max_index = np.argmax(ccf)
+    return ccf_times[max_index], ccf[max_index]
+
+
+def estimate_time_delay_ccond_av_max(
+    x: np.ndarray,
+    x_t: np.ndarray,
+    y: np.ndarray,
+    min_threshold: float = None,
+    max_threshold: float = None,
+    delta: float = None,
+    window: bool = False,
+):
+    """Estimates the average time delay by finding the time lag that maximizes
+    the cross conditional average of signal x when signal y is larger than
+    threshold. Returns also the cross conditional variance at this maximum, and
+    number of conditional averaged events.
+
+    Input:
+        x: Signal to be conditionally averaged
+        x_t: Time base of signal x
+        y: Reference signal
+        min_threshold: min threshold for conditional averaged events
+        max_threshold: max threshold for conditional averaged events
+        delta: If window = True, delta is the minimal distance between two peaks.
+        window: [bool] If True, delta also gives the minimal distance between peaks.
+
+    Returns:
+        td: Estimated time delay
+        C: Cross conditional variance at a time lag td.
+        events: Number of events larger than 2.5 the mean value
```

Comment:
This might be nitpicking but I would suggest making the returned variable names consistent with the docstring